src/details/details.test.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @jest-environment ./lib/puppeteer/environment.js
+ */
+/* eslint-env jest */
+
+let browser
+let page
+let baseUrl = 'http://localhost:3000'
+
+beforeAll(async (done) => {
+  browser = global.__BROWSER__
+  page = await browser.newPage()
+  done()
+})
+
+afterAll(async (done) => {
+  await page.close()
+  done()
+})
+
+describe('/components/details', () => {
+  describe('/components/details/preview', () => {
+    it('should add to summary the button role', async () => {
+      await page.goto(baseUrl + '/components/details/preview', { waitUntil: 'load' })
+
+      const summaryRole = await page.evaluate(() => document.body.getElementsByTagName('summary')[0].getAttribute('role'))
+      expect(summaryRole).toBe('button')
+    })
+
+    it('should set the element controlled by the summary using aria-controls', async () => {
```

Comment:
It would be good to have a test for a details element that already has an existing ID, to ensure the existing ID is used instead.

Reply:
Add the `id` attribute to the details component, but what we're generating via JavaScript and using in ARIA attributes are the `<summary>` and `<div>` container `id`s; for which one can't pass any attributes at the moment - so a test won't be relevant in this cases.

Reply:
Gotcha.